plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlab.ts

```diff
@@ -38,14 +38,33 @@ export function createPublishGitlabAction(options: {
   return createTemplateAction<{
     repoUrl: string;
     defaultBranch?: string;
+    /** @deprecated in favour of settings.visibility field */
     repoVisibility?: 'private' | 'internal' | 'public';
     sourcePath?: string;
     token?: string;
     gitCommitMessage?: string;
     gitAuthorName?: string;
     gitAuthorEmail?: string;
     setUserAsOwner?: boolean;
+    /** @deprecated in favour of settings.topics field */
     topics?: string[];
+    settings?: Record<string, any>;
+    branches?: Array<{
+      name: string;
+      protect?: boolean;
+      create?: boolean;
+      ref?: string;
+    }>;
+    variables?: Array<{
```

Comment:
I think we should call this `projectVariables` to make it very clear what it actually uses under the hood.